media-libs/jbigkit/jbigkit-2.1.recipe

```diff
@@ -0,0 +1,88 @@
+SUMMARY="A lossless image compression library"
+DESCRIPTION="JBIG-KIT is a software implementation of the JBIG1 data \
+compression standard (ITU-T T.82), which was designed for bi-level image \
+data, such as scanned documents. This library is available in portable C \
+code. It is widely used in fax products, printer firmware, printer drivers, \
+document management systems and imaging software."
+HOMEPAGE="https://www.cl.cam.ac.uk/~mgk25/jbigkit/"
+COPYRIGHT="1995-2014 Markus Kuhn
+	1989, 1991 Jef Poskanzer"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://www.cl.cam.ac.uk/~mgk25/download/jbigkit-$portVersion.tar.gz"
+CHECKSUM_SHA256="de7106b6bfaf495d6865c7dd7ac6ca1381bd12e0d81405ea81e7f2167263d932"
+PATCHES="jbigkit-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86_gcc2 x86"
+
+libVersion=$portVersion
+libVersionCompat="$libVersion compat >= 2"
+
+PROVIDES="
+	jbigkit$secondaryArchSuffix = $portVersion
+	cmd:jbgtopbm$secondaryArchSuffix = $portVersion
+	cmd:jbgtopbm85$secondaryArchSuffix = $portVersion
```

Comment:
```diff
 PROVIDES="
 	jbigkit$secondaryArchSuffix = $portVersion
-	cmd:jbgtopbm$secondaryArchSuffix = $portVersion
-	cmd:jbgtopbm85$secondaryArchSuffix = $portVersion
-	cmd:pbmtojbg$secondaryArchSuffix = $portVersion
-	cmd:pbmtojbg85$secondaryArchSuffix = $portVersion
 	lib:libjbig$secondaryArchSuffix = $libVersionCompat
 	lib:libjbig85$secondaryArchSuffix = $libVersionCompat
 	"
+if [ -z "$secondaryArchSuffix" ]; then
+PROVIDES="$PROVIDES
+	cmd:jbgtopbm = $portVersion
+	cmd:jbgtopbm85 = $portVersion
+	cmd:pbmtojbg = $portVersion
+	cmd:pbmtojbg85 = $portVersion
+	"
+fi
```

Reply:
Question :) What if I only install the secondaryArch package? the cmd's are not installed then?

Reply:
Yeah, that's a good question.

There are 4 scenari for recipes which build on all primary architectures
1) We install everything, on both primary and secondary arch.
2) We install everything on primary architectures only, and on secondary arch we only install the libs. This is OK when speed is not important when running the `cmd:` runtimes.
3) On x86_gcc2 primary arch we only install the lib, and we rename the runtimes from the secondary arch to drop the x86. This is nice if speed is important, or if the runtimes compiled with gcc2 don't work as well as those compiled with gcc5. Otherwise, imo, this is not as nice as scenario 2.
4) Same as 3) but without the renaming.
5) A "mix" of 1) and 3) but moving the `cmd:` to a subpackage. That's what I did for zstd. (I did so because one of the runtimes did not build with gcc2.)

I'd say that if speed is not important then scenario 2 is the best choice because the recipe does not have to make any test about a specific arch. So the recipe is quite simple with this scenario. But I'm OK with any other scenario.

Reply:
I have to admit that scenario 1 has a big advantage: simplicity. So I'm OK with it, unless other contributors think scenario 2 is more appropriate for this recipe.